packages/mlocate.rb

```diff
@@ -2,35 +2,39 @@
 
 class Mlocate < Package
   description 'mlocate is a locate/updatedb implementation.'
-  homepage 'https://pagure.io/mlocate'
-  version '0.26-1'
+  homepage 'https://pagure.io/mlocate/'
+  version '0.26-2'
   license 'GPL-2'
   compatibility 'all'
   source_url 'https://releases.pagure.org/mlocate/mlocate-0.26.tar.xz'
   source_sha256 '3063df79fe198fb9618e180c54baf3105b33d88fe602ff2d8570aaf944f1263e'
 
-  binary_url ({
-    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mlocate/0.26-1_armv7l/mlocate-0.26-1-chromeos-armv7l.tar.xz',
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mlocate/0.26-1_armv7l/mlocate-0.26-1-chromeos-armv7l.tar.xz',
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mlocate/0.26-1_i686/mlocate-0.26-1-chromeos-i686.tar.xz',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/mlocate/0.26-1_x86_64/mlocate-0.26-1-chromeos-x86_64.tar.xz',
-  })
-  binary_sha256 ({
-    aarch64: '69e02f44a498cd38a8f4bc2e3410f934506426d670664c402a09bfb34d51418c',
-     armv7l: '69e02f44a498cd38a8f4bc2e3410f934506426d670664c402a09bfb34d51418c',
-       i686: '74acf4453947b3570f68c43e485cb8e9f2da688e4a50336d593bbba1b6ebd2ee',
-     x86_64: 'f943fe00ad4666f5a9c7bf12d710cd8f2e136a33ad94568863b92dd45526d7eb',
-  })
+  def self.patch
+      @updatedb_wrapper = <<~EOF
+      #!/bin/bash -e
+      if [ "${EUID}" = 0 ]; then
+        echo "updatedb should not be run as root on ChromeOS."
+        exit 1
+      else
+        #{CREW_PREFIX}/bin/updatedb.real "${@}"
+      fi
+    EOF
+    IO.write("updatedb.sh", @updatedb_wrapper)
+  end
 
   def self.build
-    system "mkdir -p #{CREW_PREFIX}/db/mlocate"
-    system "sed -i 's,\$(localstatedir)/,#{CREW_PREFIX}/db/,g' Makefile.*"
-    system "sed -i \"s/groupname = mlocate/groupname = #{USER}/g\" Makefile.*"
-    system "./configure"
-    system "make"
+    system "sed -i 's:groupname = mlocate:groupname = #{USER}:g' Makefile.*"
+    system "#{CREW_ENV_OPTIONS} ./configure #{CREW_OPTIONS}"
+    system 'make'
   end
 
   def self.install
     system "make", "DESTDIR=#{CREW_DEST_DIR}", "install"
+    FileUtils.mv "#{CREW_DEST_PREFIX}/bin/updatedb", "#{CREW_DEST_PREFIX}/bin/updatedb.real"
```

Comment:
I don't think this is a good idea.  We should probably just remove the binaries from findutils or remove the package altogether.

Reply:
That is how Debian does it. It would be nice to have updated versions of `find` and `xargs`, so let's not remove the `findutils` package. Also, when I finally get to working on the `crew-alternatives` package, our own little implementation of `dpkg`'s `update-alternatives`, we can offer `findutils`'s implementation as an option.

Reply:
I think our target is making them exist at the system if possible but not keeping once only, a better option is to isolate them from /use/local filesystems and make symlinks to /usr/local like homebrew does

Reply:
There are a lot of things I don't like about Homebrew including excessive symlinks and the overuse of non-standard PATH locations.  This trap is something I want to get away from in Chromebrew.

Reply:
@supechicken, I have a solution that will satisfy both you and @uberhacker, but that's out of the scope of this PR.